src/packet/superpacket.cpp

```diff
@@ -19,30 +21,43 @@ void SuperPacket::print_packet(FILE *out) {
     fprintf(out, "}\n");
 }
 
-SuperPacket::SuperPacket(void *pkt, uint32_t max_payload_len) {
+SuperPacket::SuperPacket(void *pkt, uint32_t max_payload_len, Config *c) {
+    struct radiotap_header *radiotaph;
+    struct wlan_header * wlanh;
+
     struct ip *ipv4h;
     struct ether_header *eth;
 
+    this->config = c;
+
     parseable = true;
 
-    this->max_payload_len = max_payload_len;
-    eth = (struct ether_header *)pkt;
+    if (c->wireless == 1) {
```

Comment:
how can we just programmatically check here?

Reply:
Resolved